### PR TITLE
Remove event delegation on detached nodes

### PR DESCRIPTION
**Pull request by j3tan**
- fixes a rare edge case where a node is detached during mouseleave,
  but mouseout event is still fired on the node

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @j3tan, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by nzakas** [comment]
Is this change testable? Also, don't forget to sign the CLA. :)

**Comment by j3tan** [comment]
I couldn't think of a good test strategy for this since its kind of like a race condition with browser events.

**Comment by j3tan** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @j3tan has just signed the CLA. Thanks, and we look forward to your contribution.

**Comment by nzakas** [comment]
Can you add some comments in the code describing this edge case? I'm still not sure I understand what is happening and I can see us forgetting over time.

**Comment by j3tan** [comment]
Here's a fiddle that represents this edge case: http://jsfiddle.net/fjgjb2bh/

This happens when you have things like 'edit-in-place' that show and hide automatically.

File: tests/application-test.js [hunk 1]
```diff
@@ -486,6 +486,23 @@ describe('Box.Application', function() {
 			});
 		});
 
+		it('should still be called with a null element when an event occurs on a recently detached element', function() {
+			// Background on this edge case:
```
Comment (nzakas) [inline tests/application-test.js]: Nice, can you also put this in getNearestElement() to explain why adding `element` to the condition is important?

Comment (j3tan) [inline tests/application-test.js]: sure
